src/components/pipelines-overview/list-pages/PipelineRunsForPipelinesList.tsx

```diff
@@ -1,92 +1,106 @@
 import * as React from 'react';
-import { EmptyState, EmptyStateVariant } from '@patternfly/react-core';
 import { useTranslation } from 'react-i18next';
-import { sortable } from '@patternfly/react-table';
+import { EmptyState, EmptyStateVariant } from '@patternfly/react-core';
+import { classNames, sortable } from '@patternfly/react-table';
 import {
   TableColumn,
   VirtualizedTable,
+  useActiveColumns,
 } from '@openshift-console/dynamic-plugin-sdk';
-import { mainDataType } from '../utils';
+import { SummaryProps, sortByNumbers, sortByProperty, sortTimeStrings } from '../utils';
 import PipelineRunsForPipelinesRow from './PipelineRunsForPipelinesRow';
 
 type PipelineRunsForPipelinesListProps = {
-  mainData?: mainDataType[];
+  summaryData: SummaryProps[];
+  summaryDataFiltered?: SummaryProps[];
 };
 
 const PipelineRunsForPipelinesList: React.FC<
   PipelineRunsForPipelinesListProps
-> = ({ mainData }) => {
+> = ({ summaryData, summaryDataFiltered }) => {
   const { t } = useTranslation('plugin__pipeline-console-plugin');
   const EmptyMsg = () => (
     <EmptyState variant={EmptyStateVariant.large}>
       {t('No PipelineRuns found')}
     </EmptyState>
   );
-  const tableColumnClasses = ['', '', '', '', '', '', ''];
 
-  const columns = React.useMemo<TableColumn<mainDataType>[]>(
+  const tableColumnClasses = [
```

Comment:
Can you move it to a common file and use it all the places as we are repeating the same code at multiple places?